Assets/MixedRealityToolkit-SDK/Features/Input/FocusProvider.cs

```diff
@@ -220,34 +198,81 @@ public override int GetHashCode()
             }
         }
 
-        #region MonoBehaviour Implementation
+        #region IMixedRealityService Implementation
 
-        private async void Awake()
+        /// <inheritdoc />
+        public override void Initialize()
         {
             if (uiRaycastCamera == null)
             {
-                Debug.LogWarning("No UIRaycastCamera assigned! Creating a new UIRaycastCamera.\n" +
-                                 "To create a UIRaycastCamera in your scene, find this Focus Provider GameObject and add one there.");
-                CreateUiRaycastCamera();
+                EnsureUiRaycastCameraSetup();
             }
 
-            await WaitUntilInputSystemValid;
+            if (MixedRealityToolkit.InputSystem == null)
+            {
+                Debug.LogError($"Unable to start {Name}. An Input System is required for this feature.");
+                return;
+            }
+
+            if (MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile == null)
+            {
+                Debug.LogError($"Unable to start {Name}. An Input System Profile is required for this feature.");
+                return;
+            }
+
+            if (MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile.FocusProfile == null)
+            {
+                Debug.LogError($"Unable to start {Name}. An Focus Profile is required for this feature.");
+                return;
+            }
 
             foreach (var inputSource in MixedRealityToolkit.InputSystem.DetectedInputSources)
             {
                 RegisterPointers(inputSource);
             }
         }
 
-        private void Update()
+        /// <inheritdoc />
+        public override void Update()
         {
-            if (MixedRealityToolkit.InputSystem == null) { return; }
+            if (MixedRealityToolkit.InputSystem == null)
```

Comment:
Can we break these calls in to a generic function, I hate duplicated code :D

Reply:
Same comment as above